controllers/directivebreakdown_controller.go

```diff
@@ -421,12 +421,17 @@ func (r *DirectiveBreakdownReconciler) populateStorageBreakdown(ctx context.Cont
 		var lustreComponents []lustreComponentType
 		lustreComponents = append(lustreComponents, lustreComponentType{"AllocateAcrossServers", breakdownCapacity, "ost", ""})
 
+		mdtColocationKey := ""
+		if nnfStorageProfile.Data.LustreStorage.ExclusiveMDT {
+			mdtColocationKey = "lustre-mdt"
```

Comment:
The current use of our exclusive colocation constraint for an allocation set looks like this:
```
colocation:
- key: lustre-mgt
  type: exclusive
```

That means that this allocation set can't be colocated with any other allocation sets that have the same exclusive colocation constraint key. Practically, that allows the MGTs to exist on different nodes, but lets them be shared with OSTs and MDTs both from this filesystem and others. What you have here would do the same for MDTs. Two MDTs couldn't be on the same node, but an MDT could be colocated with one or more OSTs. I don't think that's what you want.

If you wanted the MDT to not colocate with an OST or MGT, then each of those allocation sets would need an exclusive colocation constraint that shared the same key. However, that would mean that the OSTs would now be restricted to only one OST per server. That might not be what the performance team wants.

Also, the exclusive colocation constraints would only apply for Lustre targets from that single file system. If another Lustre filesystem was allocated without the exclusive_mdt option, then it wouldn't have the colocation constraints in the directive breakdown, and it could place Lustre targets on the servers with the MDTs from the first file system. Maybe that's desirable, maybe not? The performance testers will probably only make a single file system at a time, so I don't think it matters for their use case.

I think a better option would be to add an exclusive colocation constraint without a key. That would mean that the allocation set was exclusive with everything, not just scoped to allocation sets with the same key. The Flux team would have to be told how to interpret that. If it was valuable, we could have a "within" field that scoped it to either the current filesystem or to all file systems. Doing it this way would solve the problem of having the MDTs be exclusive while still allowing more than one OST on a Rabbit.

```
    allocationSets:
    - allocationStrategy: AllocateAcrossServers
      constraints:
        labels:
        - dws.cray.hpe.com/storage=Rabbit
      label: ost
      minimumCapacity: 50000000000
    - allocationStrategy: AllocateSingleServer
      constraints:
        colocation:
        - type: exclusive
          within: directive
        labels:
        - dws.cray.hpe.com/storage=Rabbit
      label: mdt
      minimumCapacity: 5368709120
    - allocationStrategy: AllocateSingleServer
      constraints:
        colocation:
        - key: lustre-mgt
          type: exclusive
        labels:
        - dws.cray.hpe.com/storage=Rabbit
      label: mgt
      minimumCapacity: 1073741824
```